java/docker-build/Dockerfile

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright (c) 2025, NVIDIA CORPORATION. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+###
+# Build the image for spark-rapids-jni development environment.
+#
+# Arguments: CUDA_VERSION=[12.X.Y], OS_RELEASE=[8, 9], TARGETPLATFORM=[linux/amd64, linux/arm64]
+#
+###
+ARG CUDA_VERSION=12.9.1
+ARG OS_RELEASE=9
+ARG TARGETPLATFORM=linux/amd64
+
+# multi-platform build with: docker buildx build --platform linux/arm64,linux/amd64 <ARGS> on either amd64 or arm64 host
+# check available official arm-based docker images at https://hub.docker.com/r/nvidia/cuda/tags (OS/ARCH)
+FROM --platform=$TARGETPLATFORM nvidia/cuda:$CUDA_VERSION-devel-rockylinux$OS_RELEASE
+
+# If DEV_BUILD is ON, the gcc-toolset will be enabled by default for bash shell
+ARG DEV_BUILD=OFF
+
+# Dependency versions
+# Act as default GCC toolset in the image
+ARG TOOLSET_VERSION=14
+ARG CMAKE_VERSION=3.30.4
+ARG CCACHE_VERSION=4.11.2
+
+# Default x86_64 from x86 build, aarch64 cmake for arm build
+ARG CMAKE_ARCH=x86_64
+
+### Install basic requirements
+RUN dnf --enablerepo=crb install -y scl-utils gcc-toolset-${TOOLSET_VERSION} python39 \
```

Comment:
The dnf command will need to change based on the OS version.

For Rocky 8 you want `--enablerepo=devel` instead of crb

Reply:
Fixed with a conditional.